### ISSUE TITLE
Epoch number not right

### ISSUE THREAD

**Issue by AnnyShen55** [huggingface/trl#1948]
I have:
        --max_seq_length 1024 \
        --max_steps 10000 \
        --per_device_train_batch_size 1 \
        --gradient_accumulation_steps 8 \
        --learning_rate 2e-5 \
        --logging_steps 10\
        --warmup_steps 20 \
        --num_proc "$(nproc)"
as the input of SFTTrainer, since i have 8 gpus, the logging should happen passing every 80 samples. However, the epoch shows up in the logging as if between each step, 640 samples are passed. Any ideas? (I think the gradient_accumulation_steps are not related to the step count?)

**Comment by qgallouedec** [comment]
8 GPU, 8 accumulation steps, logging every 10 step -> 8*8*10 = 640 samples between logging steps

**Comment by AnnyShen55** [comment]
> 8 GPUs, 8 accumulation steps, logging every 10 step -> 8x8x10 = 640 samples between logging steps

I think the accumulation step is not related to logging step?

**Comment by qgallouedec** [comment]
Let me rephrase: 

The number of samples used per step is:

\[ \text{Number of GPUs} \times \text{Number of Gradient Accumulations} \times \text{Batch Size per Device} \]

So, if your per-device batch size is 1, the number of gradient accumulations is 8, and the number of GPUs is 8, then 1 step is made of 64 samples.

**Comment by AnnyShen55** [comment]
Understood. Thank you for your timely response!
